sky/utils/log_utils.py

```diff
@@ -1,16 +1,60 @@
 """Logging utils."""
 import enum
+import threading
 from typing import List, Optional
 
+import rich.console as rich_console
+
 import colorama
 import pendulum
 import prettytable
-import rich.status
 
 from sky import sky_logging
 
 logger = sky_logging.init_logger(__name__)
 
+console = rich_console.Console()
+_status = None
+
+
+class NoOpConsoleStatus:
+    """An empty class for multi-threaded console.status."""
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def update(self, text):
+        pass
+
+    def stop(self):
+        pass
+
+    def start(self):
+        pass
+
+
+def safe_rich_status(msg: str):
+    """A wrapper for multi-threaded console.status."""
+    if (threading.current_thread() is threading.main_thread() and
+            not sky_logging.is_silent()):
+        global _status
+        if _status is None:
+            _status = console.status(msg)
+        _status.update(msg)
+        return _status
+    return NoOpConsoleStatus()
+
+
+def force_update_rich_status(msg: str):
+    """A wrapper for multi-threaded console.status."""
```

Comment:
Shall we say the difference vs. L40?

Reply:
Oops, I forgot to update the docstr. Updated. Thanks!